app/models/identity.rb

```diff
@@ -9,8 +9,10 @@ def self.find_with_omniauth(auth)
   def self.create_with_omniauth(auth)
     Rails.logger.debug('Creating user from')
     Rails.logger.debug(auth)
-    user = User.create_with_omniauth(auth.info, auth.provider)
-    Rails.logger.debug(user.inspect)
-    create(uid: auth['uid'], provider: auth['provider'], user_id: user.id)
+    transaction do
+      user = User.create_with_omniauth(auth.info, auth.provider)
+      Rails.logger.debug(user.inspect)
+      create!(uid: auth['uid'], provider: auth['provider'], user_id: user.id)
+    end
```

Comment:
As we're writing to two tables, I wrapped it in a transaction to ensure data integrity.